lib/session/sessionManager.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { OnSessionMessage, SessionHandlerOptions, MessageSession } from "./messageSession";
+import { OnError } from "../core/messageReceiver";
+import { ClientEntityContext } from "../clientEntityContext";
+import { getProcessorCount } from "../util/utils";
+import * as log from "../log";
+
+export class SessionManager {
+  /**
+   * @property {boolean} isManagingSessions Indicates whether the manage is currently managing
+   * sessions.
+   * - **Default**: `false`.
+   */
+  isManagingSessions: boolean = false;
+  /**
+   * @property {number} maxConcurrentSessions The maximum number of sessions that the user wants to
+   * handle concurrently.
+   * - **Default**: `2000`.
+   */
+  set maxConcurrentSessions(value: number) {
+    if (value <= 0) {
+      throw new Error("'maxConcurrentSessions must be greater than 0.");
+    }
+    this._maxConcurrentSessions = value;
+    this._maxConcurrentAcceptSessionRequests = value;
+  }
+  get maxConcurrenSessions(): number {
+    return this._maxConcurrentSessions;
+  }
+  private _maxConcurrentSessions: number;
+  /**
+   * @property {number} _maxConcurrentAcceptSessionRequests The maximum number of acceptSession
+   * requests that can be made concurrently at any given time.
+   */
+  private set _maxConcurrentAcceptSessionRequests(value: number) {
+
+    this._maxConcurrentAcceptSessionRequests = Math.min(value, getProcessorCount());
```

Comment:
Wont this result in an infinite setter because the name `_maxConcurrentAcceptSessionRequests` is used both for the setter and what is being set inside?

Reply:
That is true. I am aware of it. Will be updating this file in a day or two. Have been working on SessionManager locally. Haven't updated those changes yet.